src/ngraph/op/topk.cpp

```diff
@@ -277,20 +280,26 @@ void op::v1::TopK::validate_and_infer_types()
                                       get_input_element_type(1));
     }
 
+    if (m_axis >= 0)
+    {
+        m_normalized_axis = m_axis;
+    }
+    else
+    {
+        NODE_VALIDATION_CHECK(this,
```

Comment:
Just wondering - can we postpone the normalization until the shape is known? Perhaps we can move it further to the `if` that checks if the `output_shape` rank is static? If we perform this check here, we won't be able to construct a graph that might still be valid when static shapes are passed in.

Reply:
I agree with you. Axis is used to calculate output shape only for static rank case, so restrictions can be reduced. As a result `get_axis` and `set_axis` must be modified (in current version exception can be thrown in `get_axis`)